### ISSUE TITLE
Does augmentation (such as flipud, fliplr, resize) be considered in anchor generation process ?

### ISSUE THREAD

**Issue by pocketvision** [ultralytics/yolov5#1901]

**Comment by glenn-jocher** [comment]
@daikankan no, the augmentation policy is not executed on labels loaded for autoanchor. Label augmentation is intrinsically tied to image augmentation in random_perspective(), they are thus inseparable in the current dataloader. In any case its unclear if we would want to evolve anchors on augmented labels, as our metric is mAP on the (unaugmented) validation set.

In your example flipud and fliplr will naturally have zero effect on anchor shapes, but scale will. Scale can be manually approximated by uncommenting this line:
https://github.com/ultralytics/yolov5/blob/1d1c0567a415941c109d1a2bf1a10ffde319f937/utils/autoanchor.py#L116

**Comment by jaqub-manuel** [comment]
Dear @glenn-jocher 

Yo said thatt Flipud and fliplr effect is 0.

So what data augmentation techniques have an effect on autoanchor? For example, autoanchor is not calculated in methods such as sclae, shear, mosaic and mixup? Autoanchor has no effect on values like rotation, hsv, flip, saturation only?

If so, wouldn't it make more sense to find new anchor values using properties like scale and shear using the autoanchor method? Then, training is started with the normal training set.

**Comment by glenn-jocher** [comment]
@jaqub-manuel this is more of a research oriented question. I'd suspect you would want anchors aligned with the image-space you want to deploy to, which may or may not include augmentation effects depending on your domain and use case etc. Feel free to experiment.

**Comment by abhiagwl4262** [comment]
@glenn-jocher 
Before calculating anchors, you are changing width and height as per the required image dimension (imgsz) using following line of code. 
https://github.com/ultralytics/yolov5/blob/1d1c0567a415941c109d1a2bf1a10ffde319f937/utils/autoanchor.py#L108

That means anchors are being calculated on resized images

**Comment by glenn-jocher** [comment]
@abhiagwl4262 yes anchors are defined as a function of training image sizes.

**Comment by leeyunhome** [comment]
> @abhiagwl4262 yes anchors are defined as a function of training image sizes.

Hello,

Is it not necessary to manually calculate the anchor box when training with a custom dataset?

Thank you.

**Comment by abhiagwl4262** [comment]
@leeyunhome Custom Anchors always seems to be giving better results

**Comment by glenn-jocher** [comment]
@leeyunhome autoanchor will evolve new anchors custom suited to your dataset and training parameters if your existing anchors are deemed to be poor fits to your data. No action is required on your part, this is all part of the default YOLOv5 training workflow.

**Comment by leeyunhome** [comment]
> @leeyunhome autoanchor will evolve new anchors custom suited to your dataset and training parameters if your existing anchors are deemed to be poor fits to your data. No action is required on your part, this is all part of the default YOLOv5 training workflow.

Are you saying it has nothing to do with inferencing accuracy?

Thank you.

**Comment by glenn-jocher** [comment]
@leeyunhome no I'm not saying that. Anchors are important for good accuracy.
